### PR TITLE
Remove blocked convs on incremental sync

### PR DESCRIPTION
**Pull request by mlsteele**
Blocked conversations were showing up unexpectedly. In the following scenario: Devices X and Y have the inbox loaded. Device X goes offline. Device Y blocks a conversation. Device X comes online. An incremental sync would occur that pushed the blocked conv to the gui.

This patch makes it so that in an incremental sync, blocked convs are omitted from `items` and get put into a `removals` list. The gui deletes those convs from the metamap.

### CODE REVIEW

**Comment by chrisnojima** [comment]
can't we just do a full sync? blocking barely happens so i'm not worried about the perf. diff should be a lot smaller

**Comment by mlsteele** [comment]
That would lead to full refreshes if some conv is stuck the removal list on every sync. Which could be considered a bug, but pretty hard to rule out, and not necessarily even a bug since service and gui sync different sets of convs. Does a full sync cause the screen to wipe? Anyway this should be robust against all that

**Comment by zanderz** [comment]
I tried to test it, and it seemed to make a difference, but not sure I duplicated the intended case.

**Comment by mlsteele** [comment]
good?

File: shared/actions/chat2/index.js [hunk 1]
```diff
@@ -594,9 +594,10 @@ const onChatInboxSynced = (state, action) => {
         }
         return arr
       }, [])
+      const removals = syncRes.incremental && syncRes.incremental.removals || []
```
Comment (buoyad) [inline shared/actions/chat2/index.js]: `syncRes.incremental?.removals || []`

File: shared/actions/json/chat2.json [hunk 2]
```diff
@@ -115,6 +115,7 @@
     // Got some new inbox rows
     "metasReceived": {
       "metas": "Array<Types.ConversationMeta>",
+      "removals?": "Array<string>", // convIDs to remove
```
Comment (buoyad) [inline shared/actions/json/chat2.json]: This should be `Types.ConversationIDKey` and converted in the saga instead

**Pull request merged**